concourse/steps/release.py

```diff
@@ -274,6 +274,10 @@ def apply(self):
         next_cycle_commit = self._add_all_and_create_commit(
             message=self._next_dev_cycle_commit_message(next_cycle_dev_version)
         )
+
+        if not next_cycle_commit:
```

Comment:
to me, this indicates that this step does too many things.

instead of doing an early-exit here, rather split the transaction-step into multiple parts:

- one step doing the rebase (if configured)
  - note: if no rebase to be done, do not add the step that would to the rebase. do not add an early exit to the rebase-step
- one step creating the release-commit (if required according to configuration)
- one step creating the "next version bump" (if required according to configuration)